### PR TITLE
Cleanup Oauth code

### PR DESCRIPTION
**Pull request by dialelo**
- [x] Implemented OAuth code in terms of user algebra instead of repository, moving the `getOrCreate` function to the `UserOps` primitives.
- [x] Cleaned up code in the OAuth controller, separating GitHub communication in their own methods. In the future we may want to split this further into a propper GH client, when we use GitHub data for showing contributors.
- [x] Improved the JSON parsing, it will no longer fail at runtime.
- [x] Changed OAuth scope so we have only access to data that is public

@raulraja @rafaparadela @juanpedromoreno take a look and let me know if something is missing.

### CODE REVIEW

File: site/server/app/com/fortysevendeg/exercises/utils/OAuth2.scala [hunk 1]
```diff
@@ -34,79 +40,108 @@ object OAuth2 {
...
+    (JsPath \ "avatar_url").read[String] and
+    (JsPath \ "html_url").read[String] and
+    (JsPath \ "email").readNullable[String]
+  )(GitHubUser.apply _)
+
+  def githubUserRequest(authToken: String) =
+    ws.url("https://api.github.com/user").withHeaders(HeaderNames.AUTHORIZATION → s"token $authToken").get()
+
+  def fetchGitHubUser(authToken: String): Future[Option[GitHubUser]] = for {
+    response ← githubUserRequest(authToken)
```
Comment (juanpedromoreno) [inline site/server/app/com/fortysevendeg/exercises/utils/OAuth2.scala]: Maybe we could change this for comprehension for a single `map`.

**Comment by juanpedromoreno** [comment]
Great work! Minor comment added and LGTM! 

I was wondering if it makes sense to move the `Github` stuff to a different algebra in  a different PR (oauth, contributors, etc. services in an isolated algebra).

**Comment by rafaparadela** [comment]
Probably @juanpedro is right, but we're are thinking on making our own
github client isolated in other repo (there isn't any maintained github
client in Scala). So all these algebras that Juan Pedro mentioned would be
so simple.

**Comment by rafaparadela** [comment]
Btw: looks good to me

**Comment by dialelo** [comment]
> I was wondering if it makes sense to move the Github stuff to a different algebra in a different PR (oauth, contributors, etc. services in an isolated algebra).

Maybe, as @rafaparadela suggests, we can address this as part of #170

**Comment by juanpedromoreno** [comment]
Sure :+1:

**Pull request merged**